src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/RestrictedPatientsService.kt

```diff
@@ -99,63 +103,50 @@ class RestrictedPatientsService(
     if ("REL" != latestMovement.movementType) {
       throw BadRequestException(errorCode = "LAST_MOVE_NOT_REL", message = "Prisoner ($offenderNo) was not released")
     }
-    latestMovement.fromAgency ?: throw BadRequestException(errorCode = "LAST_MOVE_NO_FROM_AGENCY", message = "Prisoner ($offenderNo) does not have an agency id associated with the last movement")
+    latestMovement.fromAgency ?: throw BadRequestException(
+      errorCode = "LAST_MOVE_NO_FROM_AGENCY",
+      message = "Prisoner ($offenderNo) does not have an agency id associated with the last movement",
+    )
   }
 
-  private fun getExistingRestrictedPatientDischargeData(latestMovement: MovementResponse, offenderNo: String): ExistingDischargeData {
-    val dischargeDateTime = calculateDischargeDateTime(offenderNo, latestMovement.movementDate, latestMovement.movementTime)
+  private fun getExistingRestrictedPatientDischargeData(
+    latestMovement: MovementResponse,
+    offenderNo: String,
+  ): ExistingDischargeData {
+    val dischargeDateTime =
+      calculateDischargeDateTime(offenderNo, latestMovement.movementDate, latestMovement.movementTime)
     return ExistingDischargeData(
       dischargeTime = dischargeDateTime,
       fromLocationId = latestMovement.fromAgency!!,
       comment = latestMovement.commentText,
     )
   }
 
-  private fun calculateDischargeDateTime(offenderNo: String, movementDate: String?, movementTime: String?): LocalDateTime {
+  private fun calculateDischargeDateTime(
+    offenderNo: String,
+    movementDate: String?,
+    movementTime: String?,
+  ): LocalDateTime {
     try {
       return LocalDateTime.parse("${movementDate}T$movementTime")
     } catch (e: Exception) {
-      throw BadRequestException(errorCode = "DISCHARGE_TIME_INVALID", message = "Prisoner ($offenderNo) does not have a valid movement date/time")
-    }
-  }
-
-  private fun addRestrictedPatient(restrictedPatient: RestrictedPatient): RestrictedPatientDto {
-    val newRestrictedPatient = restrictedPatientsRepository.saveAndFlush(restrictedPatient)
```

Comment:
this is now changed to a `save` since it will be committed by the transaction before we need to next read the data (by the indexer)